pingcap/tidb-operator
# Support pass raw toml config for pump as string

### What problem does this PR solve? 
Support pass raw toml config for pump as string

issue in #3303 still exist if config in the origin way.
### What is changed and how does it work?

- enhance the `GenericConfig` to support unmarshal from both string and object of JSON data.
- add some useful method to get&set value of the `GenericConfig`
- avoid code the json tag inside the `GenericConfig` map, so we can use it with a specified field name if need later.

### Check List 

Tests 

 
 - Manual test (add detailed scripts or steps below)
test with:
<img width="522" alt="Screen Shot 2020-09-24 at 12 01 06 PM" src="https://user-images.githubusercontent.com/1681864/94104342-5cca3b80-fe69-11ea-99ac-c585a1d9663d.png">
check cm and pump works
<img width="554" alt="Screen Shot 2020-09-24 at 12 00 40 PM" src="https://user-images.githubusercontent.com/1681864/94104359-63f14980-fe69-11ea-8dc4-98b65f8e284e.png">

also, test with the origin yaml way.




Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
Support passing raw toml config for the pump as a string
```
